eth/api.go

```diff
@@ -1501,13 +1503,14 @@ func (api *PublicDebugAPI) SeedHash(number uint64) (string, error) {
 // PrivateDebugAPI is the collection of Etheruem APIs exposed over the private
 // debugging endpoint.
 type PrivateDebugAPI struct {
-	eth *Ethereum
+	config *core.ChainConfig
+	eth    *Ethereum
 }
 
 // NewPrivateDebugAPI creates a new API definition for the private debug methods
 // of the Ethereum service.
-func NewPrivateDebugAPI(eth *Ethereum) *PrivateDebugAPI {
-	return &PrivateDebugAPI{eth: eth}
+func NewPrivateDebugAPI(config *core.ChainConfig, eth *Ethereum) *PrivateDebugAPI {
+	return &PrivateDebugAPI{config: config, eth: eth}
```

Comment:
I don't think that's a good idea tbh. We can definitely leave spicing up `ChainMaker` for a follow up PR, that is fine. But to me it would seem a bad idea to rework the entire internal APIs adding a config field, which is dropped in a follow up PR back to the original version. I see no reason for passing all those configs around the entire codebase and since we agreed that it was indeed not the beast idea, we might as well fix it here and now and not leave it for an eventual fix that might never actually happen.